source/fe/fe_nedelec_sz.cc

```diff
@@ -2093,12 +2093,15 @@ FE_NedelecSZ<dim, spacedim>::fill_fe_face_values(
     {
       // Now have all shape_grads stored on the reference cell.
       // Must now transform to the physical cell.
+      std::vector<Tensor<2, dim>> input(n_q_points);
       std::vector<Tensor<2, dim>> transformed_shape_grads(n_q_points);
       for (unsigned int dof = 0; dof < this->dofs_per_cell; ++dof)
         {
-          mapping.transform(make_array_view(fe_data.shape_grads[dof],
-                                            offset,
-                                            n_q_points),
+          for (unsigned int q = 0; q < n_q_points; ++q)
+            {
+              input[q] = fe_data.shape_grads[dof][offset + q];
```

Comment:
My comment from the previous PR still applies:

Couldn't you have achieved the same by just calling
```
  make_array_view (fe_data.shape_grads[dof],
                                offset+q,
                                n_q_points)
```
? It seems to me that you didn't actually need to copy the data, or what am I missing?

Reply:
I'm not sure I understand this... make_array_view is what is currently used and this leads to an implicit type conversion to DerivativeForm, whereas we need to stay with a Tensor, thus calling the right mapping.transform and avoiding the exception.

Reply:
Oh, I see. What you are saying is that `fe_data.shape_grads` is of type
```
std::vector<std::vector<DerivativeForm<1, dim, dim>>> shape_grads;
```
and so if you make it an `ArrayView`, it will be an array view of `DerivativeForm<1,dim,dim>` objects. But you want the pass an array of `Tensor<2,dim>` instead. Is this correct?

The function that is called in your patch implements the `mapping_covariant_gradient` transformation we have here as follows:
```
   * <li> @p mapping_covariant_gradient: it assumes $\mathbf u(\mathbf x) =
   * J^{-T} \hat{\mathbf  u}$ so that
   * @f[
   * \mathbf T(\mathbf x) =
   * J(\hat{\mathbf  x})^{-T} \hat{\mathbf  T}(\hat{\mathbf  x})
   * J(\hat{\mathbf  x})^{-1}.
   * @f]
```
On the other hand, the function that is currently implemented at least doesn't document what it does. The question then is: Is the correct solution to just call the other function, or should we have just implemented the `mapping_covariant_gradient` transformation in the function currently called?